### PR TITLE
Updates to few RBAC changes, defaulting secrets to k8s and values.yam…

### PR DESCRIPTION
**Pull request by hrishipx**
Signed off by : @hrishipx 

**What this PR does / why we need it**:
- Updates the RBAC changes for stork pods. 
- Updates values.yaml with the latest Portworx version. 
- Defaults the secrets provider as k8s (similar to how it is on the spec generator)

**Which issue(s) this PR fixes** (optional)
Closes #

**Special notes for your reviewer**:

### CODE REVIEW

File: charts/portworx/templates/portworx-k8s-secrets.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+metadata:
+  name: px-role
+  namespace: portworx
+rules:
+- apiGroups: [""]
+  resources: ["secrets"]
+  verbs: ["get", "list", "create", "update", "patch"]
+---
+kind: RoleBinding
+apiVersion: rbac.authorization.k8s.io/v1
```
Comment (zoxpx) [inline charts/portworx/templates/portworx-k8s-secrets.yaml]: Don't we have a global flag for the RBAC version?
 (afaik RBAC version changes depending on Kubernetes version)

File: charts/portworx/values.yaml [hunk 2]
```diff
@@ -13,20 +13,20 @@ etcdEndPoint:                         # The ETCD endpoint. Should be in the form
 clusterName: mycluster                # This is the default. please change it to your cluster name.
 usefileSystemDrive: false             # true/false Instructs PX to use an unmounted Drive even if it has a filesystem.
 usedrivesAndPartitions: false         # Defaults to false. Change to true and PX will use unmounted drives and partitions.
-secretType: none                      # Defaults to None, but can be kvdb/k8s/aws-kms/vault/ibm-kp
+secretType: k8s                      # Defaults to None, but can be kvdb/k8s/aws-kms/vault/ibm-kp
```
Comment (zoxpx) [inline charts/portworx/values.yaml]: NOTE: Comment is incorrect (Default is no longer none).

Please also ensure we are keeping the default identical to the defaults on http://install.portworx.com

File: charts/portworx/values.yaml [hunk 3]
```diff
@@ -13,20 +13,20 @@ etcdEndPoint:                         # The ETCD endpoint. Should be in the form
...
 stork: true                           # Use Stork https://docs.portworx.com/scheduler/kubernetes/stork.html for hyperconvergence.
-storkVersion: 1.2.0
+storkVersion: 2.0.1
 
 customRegistryURL:
 registrySecret:
 
 lighthouse: true
-lighthouseVersion: 1.5.0
+lighthouseVersion: 2.0.1
```
Comment (zoxpx) [inline charts/portworx/values.yaml]: Note that lighthouse on https://install.portworx.com/2.0/ is indeed 2.0.1, but it also uses an additional POD (portworx/lh-config-sync:0.2), which here we don't ask the version for.
Please make sure helm-YAML will use 2nd POD (also please let's have QA validate LH-install w/ this change)

File: charts/portworx/templates/portworx-k8s-secrets.yaml (hunk 1 above)
Comment (hrishipx) [reply charts/portworx/templates/portworx-k8s-secrets.yaml]: Good catch @zoxpx . I have used the global template for it now.

File: charts/portworx/values.yaml (hunk 2 above)
Comment (hrishipx) [reply charts/portworx/values.yaml]: Updated the comments

**Pull request merged**